deployer/src/handlers/mod.rs

```diff
@@ -287,6 +296,54 @@ pub async fn get_service_resources(
     }
 }
 
+#[instrument(skip_all, fields(%project_name, %service_name, %resource_type))]
+#[utoipa::path(
+    delete,
+    path = "/projects/{project_name}/services/{service_name}/resources/{resource_type}",
+    responses(
+        (status = 200, description = "Deletes a resource owned by a service."),
+        (status = 500, description = "Database error.", body = String),
+        (status = 404, description = "Record could not be found.", body = String),
+    ),
+    params(
+        ("project_name" = String, Path, description = "Name of the project that owns the service."),
+        ("service_name" = String, Path, description = "Name of the service.")
+    )
+)]
+pub async fn delete_service_resource(
+    Extension(persistence): Extension<Persistence>,
+    Extension(mut resource_manager): Extension<ResourceManager>,
+    Path((project_name, service_name, resource_type)): Path<(String, String, String)>,
+) -> Result<()> {
+    let service = persistence
+        .get_service_by_name(&service_name)
+        .await?
+        .ok_or_else(|| Error::NotFound("service not found".to_string()))?;
+
+    let resource_type: shuttle_common::resource::Type =
+        resource_type
+            .parse()
+            .map_err(|e: resource::ParseError| Error::Convert {
+                from: "String".to_string(),
+                to: "ResourceType".to_string(),
+                message: e.to_string(),
+            })?;
+
+    persistence
+        // TODO: replace persistence::ResourceType with shuttle_common::resource::Type
+        // or implement From<&shuttle_common::resource::Type> for persistence::ResourceType
+        .get_resource(&service.id, resource_type.clone().into())
+        .await?
+        .ok_or_else(|| Error::NotFound("resource not found".to_string()))?;
+
+    resource_manager
+        .delete_resource(&project_name, &resource_type)
+        .await
+        .map_err(|e| Error::Custom(e.into()))?;
```

Comment:
Does this also eventually delete a resource in persistence?

Reply:
OMG how did I miss that :facepalm:. Fixing...